react/features/breakout-rooms/actions.js

```diff
@@ -0,0 +1,184 @@
+// @flow
+
+import i18next from 'i18next';
+import _ from 'lodash';
+import type { Dispatch } from 'redux';
+
+import {
+    conferenceLeft,
+    conferenceWillLeave,
+    createConference,
+    getCurrentConference,
+    setRoom
+} from '../base/conference';
+import { setAudioMuted, setVideoMuted } from '../base/media';
+import { getRemoteParticipants } from '../base/participants';
+import { getConferenceOptions } from '../conference/functions';
+import { clearNotifications } from '../notifications';
+
+import { UPDATE_BREAKOUT_ROOMS } from './actionTypes';
+import {
+    JSON_TYPE_ADD_BREAKOUT_ROOM,
+    JSON_TYPE_MOVE_TO_ROOM_REQUEST,
+    JSON_TYPE_REMOVE_BREAKOUT_ROOM
+} from './constants';
+import {
+    getBreakoutRooms,
+    getCurrentRoomId,
+    getMainRoomId,
+    isInBreakoutRoom
+} from './functions';
+import logger from './logger';
+
+declare var APP: Object;
+
+/**
+ * Action to create a breakout room.
+ *
+ * @returns {Function}
+ */
+export function createBreakoutRoom() {
+    return (dispatch: Dispatch<any>, getState: Function) => {
+        const { nextIndex: index = 1, rooms } = getBreakoutRooms(getState);
+        const message = {
+            type: JSON_TYPE_ADD_BREAKOUT_ROOM,
+            subject: i18next.t('breakoutRooms.defaultName', { index }),
+            nextIndex: index + 1
+        };
+
+        dispatch({
+            type: UPDATE_BREAKOUT_ROOMS,
+            nextIndex: index + 1,
+            rooms
+        });
+        getCurrentConference(getState).sendMessage(message, 'focus');
```

Comment:
Can we do this without hardcoding the focus user part?

Reply:
Thanks. I have replaced 'focus' by 'prosody_mod_breakout_rooms' as it really does not matter what the destination is. It is just to reach the backend.